sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -56,64 +80,177 @@ class RapidsGdsStore(
         0
       }
       logDebug(s"Spilled to $path $fileOffset:${other.size} via GDS")
-      new RapidsGdsBuffer(id, fileOffset, other.size, other.meta, other.getSpillPriority,
-        other.spillCallback)
+      new RapidsGdsSingleShotBuffer(id, path, fileOffset, other.size, other.meta,
+        other.getSpillPriority, other.spillCallback)
     }
-  }
 
-  class RapidsGdsBuffer(
-      id: RapidsBufferId,
-      val fileOffset: Long,
-      size: Long,
-      meta: TableMeta,
-      spillPriority: Long,
-      spillCallback: RapidsBuffer.SpillCallback)
-      extends RapidsBufferBase(id, size, meta, spillPriority, spillCallback) {
-    override val storageTier: StorageTier = StorageTier.GDS
+    class RapidsGdsSingleShotBuffer(
+        id: RapidsBufferId, path: File, fileOffset: Long, size: Long, meta: TableMeta,
+        spillPriority: Long, spillCallback: RapidsBuffer.SpillCallback)
+        extends RapidsGdsBuffer(id, size, meta, spillPriority, spillCallback) {
 
-    override def getMemoryBuffer: MemoryBuffer = getDeviceMemoryBuffer
+      override def materializeMemoryBuffer: MemoryBuffer = {
+        closeOnExcept(DeviceMemoryBuffer.allocate(size)) { buffer =>
+          CuFile.readFileToDeviceBuffer(buffer, path, fileOffset)
+          logDebug(s"Created device buffer for $path $fileOffset:$size via GDS")
+          buffer
+        }
+      }
 
-    override def materializeMemoryBuffer: MemoryBuffer = {
-      val path = if (id.canShareDiskPaths) {
-        sharedBufferFiles.get(id)
-      } else {
-        id.getDiskPath(diskBlockManager)
+      override def copyToMemoryBuffer(srcOffset: Long, dst: MemoryBuffer, dstOffset: Long,
+          length: Long, stream: Cuda.Stream): Unit = {
+        dst match {
+          case dmOriginal: DeviceMemoryBuffer =>
+            val dm = dmOriginal.slice(dstOffset, length)
+            // TODO: switch to async API when it's released, using the passed in CUDA stream.
+            CuFile.readFileToDeviceBuffer(dm, path, fileOffset + srcOffset)
```

Comment:
Is CuFile using the system default stream here?  Wondering if we need to do any synchronization on `stream` before calling the `CuFile` interface since it's not taking the stream parameter.

Reply:
Let me check with the GDS team.